### PR TITLE
Add LeverFYActions

### PR DESCRIPTION
**Pull request by lacoop6tu**
Fixes IP-2

### CODE REVIEW

File: src/test/actions/rpc/LeverFYActions.t.sol [hunk 1]
```diff
@@ -0,0 +1,1111 @@
...
+            normalDebt,
+            _getBuyFIATSwapParams(address(dai),normalDebt)
+        );
+
+        assertGt(dai.balanceOf(address(userProxy)), userProxyInitialBalance);
+        assertEq(ERC20(fyDAI2212).balanceOf(address(fyDAI2212Vault)), vaultInitialBalance);
+        assertEq(_collateral(address(fyDAI2212Vault), address(userProxy)), initialCollateral);
+    }
+
+    function test_underlierToPToken() external {
```
Comment (SolcAndMe) [inline src/test/actions/rpc/LeverFYActions.t.sol]: wrongly named test

File: src/test/actions/rpc/LeverFYActions.t.sol [hunk 2]
```diff
@@ -0,0 +1,1111 @@
...
+        assertGt(pTokenAmountNow, 0);
+        // advance some months
+        vm.warp(block.timestamp + 10 days);
+        uint256 pTokenAmountBeforeMaturity = leverActions.underlierToFYToken(100 * ONE_USDC, address(fyUSDC2212LP));
+        // closest to the maturity we get less sPT for same underlier amount
+        assertGt(pTokenAmountNow, pTokenAmountBeforeMaturity);
+
+    }
+
+    function test_pTokenToUnderlier() external {
```
Comment (SolcAndMe) [inline src/test/actions/rpc/LeverFYActions.t.sol]: wrongly named test

File: src/actions/lever/LeverFYActions.sol [hunk 3]
```diff
@@ -0,0 +1,438 @@
...
+import {WAD, toInt256, add, wmul, wdiv, sub} from "../../core/utils/Math.sol";
+import {Lever20Actions} from "./Lever20Actions.sol";
+import {IFYPool,IFYToken} from "../vault/VaultFYActions.sol";
+
+// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+// WARNING: These functions meant to be used as a a library for a PRBProxy. Some are unsafe if you call them directly.
+// !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+/// @title LeverFYActions
+/// @notice A set of vault actions for modifying positions collateralized by Yield pTokens
```
Comment (SolcAndMe) [inline src/actions/lever/LeverFYActions.sol]: Yield fyTokens

File: src/actions/lever/LeverFYActions.sol [hunk 4]
```diff
@@ -0,0 +1,438 @@
...
+    struct SellCollateralAndDecreaseLeverFlashLoanData {
+        // Address of the collateral vault
+        address vault;
+        // Address of the collateral token
+        address token;
+        // Address of the owner of the position
+        address position;
+        // Address of the account who receives the withdrawn / swapped underliers
+        address collateralizer;
+        // Amount of pTokens to withdraw and swap for underliers [tokenScale]
```
Comment (SolcAndMe) [inline src/actions/lever/LeverFYActions.sol]: replace every occurrence of pToken with fyToken

File: src/actions/lever/LeverFYActions.sol [hunk 5]
```diff
@@ -0,0 +1,438 @@
...
+    struct CollateralSwapParams {
+        // Min amount of asset out [tokenScale for buying, underlierScale for selling]
+        uint256 minAssetOut;
+        // Address of the Yield Space v2 pool
+        address yieldSpacePool;
+        // Address of the underlier (underlierToken) when buying, Address of the fyToken (token) when selling
+        address assetIn;
+        // Address of the fyToken (token) when buying, Address of underlier (underlierToken) when selling
+        address assetOut;
+    }
```
Comment (SolcAndMe) [inline src/actions/lever/LeverFYActions.sol:49]: does Yield have a deadline param? if yes we should add it.

**Review by SolcAndMe — CHANGES_REQUESTED** [review]

File: src/actions/lever/LeverFYActions.sol (hunk 4 above)
Comment (lacoop6tu) [reply src/actions/lever/LeverFYActions.sol]: updated both LeverFYActions.sol and t.sol

**Pull request merged**